evap/rewards/views.py

```diff
@@ -41,12 +41,20 @@ def redeem_reward_points(request):
     try:
         save_redemptions(request, redemptions, previous_redeemed_points)
         messages.success(request, _("You successfully redeemed your points."))
-    except (NoPointsSelected, NotEnoughPoints, RedemptionEventExpired) as error:
-        messages.warning(request, error)
-        return 400
-    except OutdatedRedemptionData as error:
-        messages.error(request, error)
-        return 409
+    except (NoPointsSelected, NotEnoughPoints, RedemptionEventExpired, OutdatedRedemptionData) as error:
+        error_string = ""
+        status_code = 400
+        if isinstance(error, NoPointsSelected):
+            error_string = "You cannot redeem 0 points."
+        elif isinstance(error, NotEnoughPoints):
+            error_string = "You don't have enough reward points."
+        elif isinstance(error, RedemptionEventExpired):
+            error_string = "Sorry, the deadline for this event expired already."
+        elif isinstance(error, OutdatedRedemptionData):
+            status_code = 409
+            error_string = "It appears that your browser sent multiple redemption requests. You can see all successful redemptions below."
+        messages.warning(request, _(error_string))
```

Comment:
In general, having a single translation call here would be a good thing. However, translation is always a bit tricky. In order to build our translation tables that store the pair <"You cannot redeem 0 points", "Du kannst nicht 0 Punkte einlösen">, we first need to find all strings in the codebase that we want to translate (-> "You cannot redeem 0 points"). Under the hood, django uses the gettext framework for this. In the end, it boils down to using a tool called [xgettext](https://www.gnu.org/software/gettext/manual/gettext.html#Template) which goes through the source code, looks for all function calls to all defined translation methods, and extracts the strings that the method is called with.

In order for that to work, the string needs to be extractable from the source code without actually running it, and thus, the method must directly be called with the string constant. So, in this case, we have to have one function call to `_` for each string literal above.